src/auth/src/credentials.rs

```diff
@@ -689,14 +689,76 @@ pub mod testing {
 mod tests {
     use super::*;
     use base64::Engine;
+    use gax::backoff_policy::BackoffPolicy;
+    use gax::retry_policy::RetryPolicy;
+    use gax::retry_result::RetryResult;
     use num_bigint_dig::BigUint;
     use reqwest::header::AUTHORIZATION;
     use rsa::RsaPrivateKey;
     use rsa::pkcs8::{EncodePrivateKey, LineEnding};
     use scoped_env::ScopedEnv;
+    use std::error::Error;
     use std::sync::LazyLock;
+    use std::sync::atomic::{AtomicBool, Ordering};
     use test_case::test_case;
 
+    #[derive(Debug)]
+    pub struct AuthRetryPolicy {
+        pub max_attempts: u32,
+    }
+
+    impl RetryPolicy for AuthRetryPolicy {
+        fn on_error(
+            &self,
+            _loop_start: std::time::Instant,
+            attempt_count: u32,
+            _idempotent: bool,
+            error: gax::error::Error,
+        ) -> RetryResult {
+            if attempt_count >= self.max_attempts {
+                return RetryResult::Exhausted(error);
+            }
```

Comment:
You should use composition for this.  Define a base policy (without attempt limits) and then use `RetryPolicyExt` to layer the retry limit.